Sources/SwiftInspectorCore/PropertyAnalyzer.swift

```diff
@@ -0,0 +1,248 @@
+      guard self.typeName == typeWithPropInfo.name else { return }
+      result = try? typeWithPropInfo.merge(with: result)
+    }
+    visitor.walk(syntax)
+    return result
+  }
+
+  // MARK: Private
+
+  private let cachedSyntaxTree: CachedSyntaxTree
+  private let typeName: String
+}
+
+// MARK: - TypeLocationSyntaxVisitor
+
+private final class TypeSyntaxVisitor: SyntaxVisitor {
+
+  init(onNodeVisit: @escaping (_ info: TypeWithPropInfo) -> Void) {
+    self.onNodeVisit = onNodeVisit
+  }
+
+  override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: ProtocolDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: ExtensionDeclSyntax) -> SyntaxVisitorContinueKind {
+    if let typeIdentifier = node.extendedType.as(SimpleTypeIdentifierSyntax.self) {
+      processNode(node, withName: typeIdentifier.name.text, members: node.members.members)
+    }
+    return .visitChildren
+  }
+
+  // MARK: Private
+
+  private let onNodeVisit: (_ info: TypeWithPropInfo) -> Void
+
+  private func processNode<Node>(_ node: Node, withName name: String, members: MemberDeclListSyntax) where Node: SyntaxProtocol {
+    var result = [TypeWithPropInfo.PropInfo]()
+    let propertyVisitor = PropertySyntaxVisitor(parentName: name) { info in
+      result.append(info)
+    }
+    propertyVisitor.walk(node)
+    onNodeVisit(.init(name: name, properties: result))
+  }
+}
+
+// MARK: - PropertySyntaxVisitor
+
+private final class PropertySyntaxVisitor: SyntaxVisitor {
+
+  init(parentName: String, onNodeVisit: @escaping (_ info: TypeWithPropInfo.PropInfo) -> Void) {
+    self.parentName = parentName
+    self.onNodeVisit = onNodeVisit
+  }
+
+  override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+    var leadingTrivia: Trivia? = node.leadingTrivia
+    var access: TypeWithPropInfo.Access = .internal
+    var scope: TypeWithPropInfo.Scope = .instance
+
+    node.modifiers?.forEach { modifier in
+      // leading trivia is on the first modifier, or the node itself if no modifiers are present
+      leadingTrivia = leadingTrivia ?? modifier.leadingTrivia
+      // This attempts to get any access information from the node's modifiers
+      if let modifierAccess = TypeWithPropInfo.Access(rawValue: modifier.name.text) {
```

Comment:
Should we be looking into DeclModifier syntax instead? I think this could break if we have an attribute like `public private(set)` since we'd be first assigning the modifier to public, but later re-assigning to private.

I think it'd be better to look at the tokens of each declaration modifier and look for the public, internal, etc keywords, then we could validate that we only have 1 token (so we make sure it's only public and not something like public(set)`?

I've found that assuming ordering when using the AST usually ends up biting me back. I think using the token names tends to be more robust and scales better.